From 095930d41e488acefb498b71afaee4666eec3f1b Mon Sep 17 00:00:00 2001
From: Michal Babej <michal.babej@intel.com>
Date: Thu, 8 May 2025 10:11:10 +0300
Subject: [PATCH] Windows: disable HOST_CPU_ENABLE_SPIRV due to mangling issues

SPIR-V on Windows is currently unusable with the CPU driver,
however this change allows to build a single PoCL with ENABLE_SPIRV
and both CPU and LZ drivers.
---
 CMakeLists.txt | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0fd93335a..0ef204ff6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1237,16 +1237,18 @@ else()
   set(ENABLE_SPIRV OFF CACHE INTERNAL "SPIR-V enabled" FORCE)
 endif()
 
-if(ENABLE_HOST_CPU_DEVICES AND ENABLE_SPIRV)
-  if(HOST_DEVICE_ADDRESS_BITS MATCHES "32")
-    set(HOST_CPU_ENABLE_SPIRV OFF)
-  else()
-    set(HOST_CPU_ENABLE_SPIRV ON)
-  endif()
+if(ENABLE_SPIRV)
   # required for the wrapper generator
   find_package(Python3 REQUIRED COMPONENTS Interpreter)
 endif()
 
+# we don't have a useful SPIRV wrapper for 32bit SPIRV
+# SPIR-V on windows is broken ATM because of mangling
+set(HOST_CPU_ENABLE_SPIRV OFF)
+if(ENABLE_HOST_CPU_DEVICES AND ENABLE_SPIRV AND (NOT WIN32) AND (HOST_DEVICE_ADDRESS_BITS MATCHES "64"))
+  set(HOST_CPU_ENABLE_SPIRV ON)
+endif()
+
 ######################################################################################
 
 set(HAVE_DLFCN_H OFF CACHE BOOL "dlopen" FORCE)
